truss/util/config_checks.py

```diff
@@ -30,6 +38,19 @@ def check_and_update_memory_for_trt_llm_builder(tr: TrussHandle) -> bool:
     return True
 
 
+def check_quantization_and_hardware_compat(tr: TrussHandle) -> bool:
+    """Check if quantization type is compatible with hardware"""
+    if tr.spec.config.trt_llm and tr.spec.config.trt_llm.build:
+        user_defined_quantization_type = tr.spec.config.trt_llm.build.quantization_type
+        user_defined_accelerator = tr.spec.config.resources.accelerator.accelerator
```

Comment:
I'm not too familiar with this aspect of pydantic, are all the fields in the tree always provided, e.g. what if there's no trt_llm section or resources section. Will we still be able to access subfield or will this fail?